### PR TITLE
fix(rust): Make read_json ignore leading BOM characters

### PR DESCRIPTION
**Pull request by deanm0000**
Fixes https://github.com/pola-rs/polars/issues/15301

### CODE REVIEW

**Comment by ritchie46** [comment]
Can you rebase? I believe tests should run then.

File: crates/polars-io/src/json/mod.rs [hunk 1]
```diff
@@ -221,6 +221,17 @@ where
     schema_overwrite: Option<&'a Schema>,
     json_format: JsonFormat,
 }
+fn remove_bom(bytes: Vec<u8>) -> Vec<u8> {
+    if bytes.starts_with(&[0xEF, 0xBB, 0xBF]) {
+        // UTF-8 BOM
+        bytes[3..].to_vec()
```
Comment (ritchie46) [inline crates/polars-io/src/json/mod.rs:227]: This wourl reallocate. We should do the before we allocate a vec from the input.

**Review by ritchie46 — COMMENTED** [review]
I left a comment.

But what I don't understand. Why are there so many unrelated changes. Can you repopen with only the json related changes?

**Comment by deanm0000** [comment]
> I left a comment.
> 
> But what I don't understand. Why are there so many unrelated changes. Can you repopen with only the json related changes?

I guess I rebased wrong. I don't use git in my day job so I've never learned it. I'll reopen in a bit.

**Comment by deanm0000** [comment]
> Can you rebase? I believe tests should run then.

**Pull request closed without merging**